pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -323,19 +332,34 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
 
         String producerName = publishContext.getProducerName();
         long sequenceId = publishContext.getSequenceId();
+        // The process of the Producer sending chunk messages is continuous, and all chunks of a message use the same
+        // message metadata and sequence ID. Therefore, it is only necessary to check if the sequence ID of the first
+        // chunk is duplicated.
+        // When we receive the initial message of a non-duplicated chunk message, we place it in the
+        // chunkMessageOngoing. Upon completion of sending this chunk message, if we receive other messages
+        // sent by this Producer, we will remove it from the chunkMessageOngoing.
+        headersAndPayload.markReaderIndex();
+        MessageMetadata msgMetadata = Commands.parseMessageMetadata(headersAndPayload);
+        headersAndPayload.resetReaderIndex();
+        if (chunkMessageOngoing.containsKey(producerName)) {
+            //Deduplication requires the dependency on the ordered nature of messages to work accurately.
+            if (publishContext.isChunked() && chunkMessageOngoing.get(producerName) < msgMetadata.getChunkId()) {
+                chunkMessageOngoing.put(producerName, msgMetadata.getChunkId());
+                return MessageDupStatus.NotDup;
```

Comment:
It appears that we don't need to check seq_id here due to the consecutive nature, but would it be safe to check it?

m1-chunk-1, seqid=1
m1-chunk-2, seqid=1
m2-chunk-1, seqid=2
m2-chunk-2, seqid=1 //are we sure this will never happen?

Reply:
I agree. It's better to check seq_id too.